Configurable cache file name

apeneve opened the pull request:
Background for pull request: I have two nuget server sites sharing the same package repository folder. One public with authentication, and one internal. In this scenario I get file lock errors on the cache file, so I need to be able to configure different cache files for each site.

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/NuGet/NuGet.Server?pullRequest=49) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/NuGet/NuGet.Server?pullRequest=49) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/NuGet/NuGet.Server?pullRequest=49) it.</sub>

joelverhagen commented:
@apeneve, sorry about the delay. Thanks for the contribution. I'm taking a look at this now!

joelverhagen commented on src/NuGet.Server/Web.config:
```diff
@@ -22,6 +22,11 @@
     <add key="packagesPath" value="" />
 
+    <!--
+    Change the name of the internal cache file. Default is machine name (System.Environment.MachineName).
```
It's unclear whether putting a path here works (or even _should_ work).

I would recommend updated the XML comment to state that this must be a file name and not a relative or absolute path. Additionally, I would validate at runtime that this is just a file name. In the future we could extend this to support relative paths but that seems unnecessary now and can be expanded without being a breaking change.

apeneve replied:
Agreed. I updated the XML comment and added validation of a valid file name.

joelverhagen commented on src/NuGet.Server.Core/Infrastructure/ServerPackageRepository.cs:
```diff
@@ -105,9 +105,28 @@ public class ServerPackageRepository
+            }
+
+            if (fileName.EndsWith(suffix, StringComparison.OrdinalIgnoreCase))
+                return fileName;
```
nitpick: curly `{` `}` around all if blocks, even if one line.

joelverhagen commented on test/NuGet.Server.Core.Tests/Infrastructure/FuncSettingsProvider.cs:
```diff
@@ -7,8 +7,8 @@ namespace NuGet.Server.Core.Tests.Infrastructure
-        readonly Func<string, bool, bool> _getSetting;
-        internal FuncSettingsProvider(Func<string,bool,bool> getSetting)
+        readonly Func<string, object, object> _getSetting;
+        internal FuncSettingsProvider(Func<string,object,object> getSetting)
```
nitpic: spaces between type parameters after the comma `,`

joelverhagen commented:
Could you add some unit or integration test that verified this behavior? I'm thinking something that verifies that if the this setting is provided, the cache file appears in the packages directory (`File.Exists` or something).

apeneve commented:
@joelverhagen  I fixed up the nitpic comments and extended the XML comment a bit. I also added some tests to the ServerPackageRepository class to describe the different the cache file name scenarios.

joelverhagen commented:
@apeneve, wonderful! I'll work on merging this and shipping this as a `3.1.0` package.

joelverhagen commented:
@apeneve, this is available as 3.1.0 on nuget.org:
https://www.nuget.org/packages/NuGet.Server/3.1.0

apeneve commented:
Excellent! I will try it out right away.

On Mon, Feb 5, 2018 at 7:02 PM, Joel Verhagen <notifications@github.com>
wrote:

> @apeneve <https://github.com/apeneve>, this is available as 3.1.0 on
> nuget.org:
> https://www.nuget.org/packages/NuGet.Server/3.1.0
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NuGet/NuGet.Server/pull/49#issuecomment-363167780>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADRLDe6cI6KMGgsIAvFuPe0k4ndPQeEsks5tR0IggaJpZM4Rq8iq>
> .
>
